Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/events/TraceLocationCreator.kt

```diff
@@ -0,0 +1,51 @@
+package de.rki.coronawarnapp.eventregistration.events
+
+import dagger.Lazy
+import de.rki.coronawarnapp.eventregistration.checkins.qrcode.TraceLocation
+import de.rki.coronawarnapp.eventregistration.checkins.qrcode.toTraceLocation
+import de.rki.coronawarnapp.eventregistration.events.server.TraceLocationServer
+import de.rki.coronawarnapp.eventregistration.storage.repo.TraceLocationRepository
+import de.rki.coronawarnapp.util.security.SignatureValidation
+import timber.log.Timber
+import javax.inject.Inject
+import javax.inject.Singleton
+
+const val TRACE_LOCATION_VERSION = 1
+
+@Singleton
+class TraceLocationCreator @Inject constructor(
+    private val api: Lazy<TraceLocationServer>,
+    private val repository: TraceLocationRepository,
+    private val signatureValidation: SignatureValidation
+) {
+
+    suspend fun createTraceLocation(traceLocationUserInput: TraceLocationUserInput): TraceLocation {
+
+        val signedTraceLocation = api.get().retrieveSignedTraceLocation(traceLocationUserInput)
```

Comment:
Where does the  trim to 100chars for address/location happen?

Reply:
Currently in the UI layer(maxLength in xml and validation in the ViewModel). What do you think, should we trust the UI layer does it correctly or should we trim it here too?

Reply:
I would trim it here too, keyboard input and EditText restrictions sometimes have quirks on different ROMs.

There was an adjustment to the techspec that we also need to prevent / disallow `\n`, a follow up PR would be best I think.

Reply:
Yes, let's do this in a follow-up PR.